pkg/apis/extensions/validation/validation.go

```diff
@@ -564,3 +569,30 @@ func ValidateClusterAutoscaler(autoscaler *extensions.ClusterAutoscaler) errs.Va
 	allErrs = append(allErrs, validateClusterAutoscalerSpec(autoscaler.Spec)...)
 	return allErrs
 }
+
+func ValidatePodSelector(ps extensions.PodSelector) errs.ValidationErrorList {
+	allErrs := errs.ValidationErrorList{}
+	allErrs = append(allErrs, apivalidation.ValidateLabels(ps.MatchLabels, "matchLabels")...)
+	for i, expr := range ps.MatchExpressions {
+		allErrs = append(allErrs, ValidatePodSelectorRequirement(expr).Prefix(fmt.Sprintf("matchExpressions.[%v]", i))...)
+	}
+	return allErrs
+}
+
+func ValidatePodSelectorRequirement(sr extensions.PodSelectorRequirement) errs.ValidationErrorList {
+	allErrs := errs.ValidationErrorList{}
+	switch sr.Operator {
+	case extensions.PodSelectorOpIn, extensions.PodSelectorOpNotIn:
+		if len(sr.Values) == 0 {
+			allErrs = append(allErrs, errs.NewFieldInvalid("operator", sr.Values, "must be empty when operator is In or NotIn"))
```

Comment:
Swap error messages with exists / does not exist. Must be non-empty here and empty below.